lambdas/initialise-ipv-session/src/main/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandler.java

```diff
@@ -319,7 +322,7 @@ private void validateAndStoreHMRCInheritedIdentity(
                 | UnrecognisedVotException
                 | AuditExtensionException e) {
             throw new RecoverableJarValidationException(
-                    OAuth2Error.INVALID_REQUEST_OBJECT.setDescription(
+                    INVALID_INHERITED_IDENTITY_ERROR_OBJECT.setDescription(
                             "Inherited identity JWT failed to validate"),
                     claimsSet,
                     e);
```

Comment:
Do we also need to catch/rethrow other `JarValidationException`/`RecoverableJarValidationException`s?

(Would we ever want to throw another error code?)

Reply:
Are you asking if there are other issues we would want to catch that would have a different description?
For inherited identity JWTs we're catching three scenarios - failure to validate, jwt is null, and more/less than 1 jwt recieved.

Reply:
Ah, I was thinking of the first case - but those will all be caught under `VerifiableCredentialException`. For some reason in my head that might throw a `JarValidationException`, but I don't think that's possible...

Reply:
We're throwing the JarValidationException if the inherited jwt is null, 0, or more than 1 (and then catching it as a recoverable.)
And catching everything else from validating the jwt itself and rethrowing.